code/game/objects/structures/lavaland/ore_vent.dm

```diff
@@ -168,35 +169,40 @@
  * May also be called after the fact, as seen in SSore_generation's initialize, to add more minerals to an existing vent.
  *
- * The above applies only when spawning in at mapload, otherwise we pick randomly from ore_vent_minerals_default.
+ * The above applies only when spawning in at mapload, otherwise we pick randomly from ore_vent_minerals_lavaland.
  *
- * @params max_minerals How many minerals should be added to this vent? Defaults to MINERAL_TYPE_OPTIONS_RANDOM, which is 4.
- * @params map_loading Is this vent being spawned in at mapload? If so, we use the ore_generation subsystem's ore_vent_minerals list to pick minerals. Otherwise, we pick randomly from ore_vent_minerals_default.
+ * @params new_minerals How many minerals should be added to this vent? Defaults to MINERAL_TYPE_OPTIONS_RANDOM, which is 4.
+ * @params map_loading Is this vent being spawned in at mapload? If so, we use the ore_generation subsystem's ore_vent_minerals list to pick minerals. Otherwise, we pick randomly from ore_vent_minerals_lavaland.
  */
-/obj/structure/ore_vent/proc/generate_mineral_breakdown(max_minerals = MINERAL_TYPE_OPTIONS_RANDOM, map_loading = FALSE)
-	if(max_minerals < 1)
-		CRASH("generate_mineral_breakdown called with max_minerals < 1.")
-	for(var/iterator in 1 to max_minerals)
-		if(!SSore_generation.ore_vent_minerals.len && map_loading)
+/obj/structure/ore_vent/proc/generate_mineral_breakdown(new_minerals = MINERAL_TYPE_OPTIONS_RANDOM, map_loading = FALSE)
+	if(new_minerals < 1)
+		to_chat(world, "CRASH, PAIN")
+		CRASH("generate_mineral_breakdown called with new_minerals < 1.")
+	var/starting_minerals = length(mineral_breakdown) // we're going to use this to track how many minerals we've added.
+	while(length(mineral_breakdown) < new_minerals + starting_minerals)
+		to_chat(world, "We are trying to go from [length(mineral_breakdown)] to [new_minerals + starting_minerals] minerals. Global list size: [length(SSore_generation.ore_vent_minerals)]")
+		if(!length(SSore_generation.ore_vent_minerals) && map_loading && !SSore_generation.initialized)
+			to_chat(world, "CRASH, PAIN. List size: [length(SSore_generation.ore_vent_minerals)]")
 			CRASH("No minerals left to pick from! We may have spawned too many ore vents in init, or added too many ores to the existing vents.")
 		var/datum/material/material
 		if(map_loading)
 			material = pick_weight(SSore_generation.ore_vent_minerals)
 		if(is_type_in_list(mineral_breakdown, material))
+			to_chat(world, "Bouncing...")
 			continue
 		if(map_loading)
+			to_chat(world, "Success.")
 			SSore_generation.ore_vent_minerals[material] -= 1 //We remove 1 from the ore vent's mineral breakdown weight, so that it can't be picked again.
 			if(SSore_generation.ore_vent_minerals[material] <= 0)
 				SSore_generation.ore_vent_minerals -= material
 		else
-			material = pick_weight(SSore_generation.ore_vent_minerals_default)
+			material = pick_weight(GLOB.ore_vent_minerals_lavaland)
```

Comment:
why do we have to check map_loading twice here? the is type in list thing doesn't do anything if it's passed null, (which is what non maploading gives you) right? (this may make this a good canidate for early continues btw)

also uh, is type in list isn't what you want here. likely want to just do like mineral_breakdown[material]

Reply:
This last pass should resolve both issues now 👍